presto-main/src/test/java/com/facebook/presto/eventlistener/TestEventListenerManager.java

```diff
@@ -0,0 +1,564 @@
+                Optional.of(failureInfo),
+                "dummyPayload");
+    }
+
+    private static SplitStatistics createDummySplitStatistics()
+    {
+        Duration cpuTime = Duration.ofSeconds(500);
+        Duration wallTime = Duration.ofSeconds(1000);
+        Duration queuedTime = Duration.ofSeconds(120);
+        Duration completedReadTime = Duration.ofSeconds(800);
+
+        long completedPositions = 1500;
+        long completedDataSizeBytes = 10000000L;
+
+        Optional<Duration> timeToFirstByte = Optional.of(Duration.ofSeconds(10));
+        Optional<Duration> timeToLastByte = Optional.empty();
+
+        return new SplitStatistics(
+                cpuTime,
+                wallTime,
+                queuedTime,
+                completedReadTime,
+                completedPositions,
+                completedDataSizeBytes,
+                timeToFirstByte,
+                timeToLastByte);
+    }
+
+    private static void tryDeleteFile(Path path)
+    {
+        try {
+            File file = new File(path.toUri());
+            if (file.exists()) {
+                Files.delete(file.toPath());
+            }
+        }
+        catch (IOException e) {
+            log.error(e, "Could not delete file found at [%s]", path);
+        }
+    }
+
+    private static class TestEventListenerFactory
+            implements EventListenerFactory
+    {
+        public static String name;
+        private final TestingEventListener testingEventListener;
+
+        public TestEventListenerFactory(TestingEventListener testingEventListener, String name)
+        {
+            this.testingEventListener = requireNonNull(testingEventListener, "testingEventListener is null");
+            this.name = name;
+        }
+
+        @Override
+        public String getName()
+        {
+            return name;
+        }
+
+        @Override
+        public EventListener create(Map<String, String> config)
+        {
+            return testingEventListener;
+        }
+    }
+
+    private static class TestingEventListener
+            implements EventListener
+    {
+        private final EventsCapture eventsCapture;
+
+        public TestingEventListener(EventsCapture eventsCapture)
+        {
+            this.eventsCapture = eventsCapture;
+        }
+
+        @Override
+        public void queryCreated(QueryCreatedEvent queryCreatedEvent)
+        {
+            eventsCapture.addQueryCreated(queryCreatedEvent);
+        }
+
+        @Override
+        public void queryCompleted(QueryCompletedEvent queryCompletedEvent)
+        {
+            eventsCapture.addQueryCompleted(queryCompletedEvent);
+        }
+
+        @Override
+        public void splitCompleted(SplitCompletedEvent splitCompletedEvent)
+        {
+            eventsCapture.addSplitCompleted(splitCompletedEvent);
+        }
+    }
+
+    private static class EventsCapture
+    {
+        private ImmutableList.Builder<QueryCreatedEvent> queryCreatedEvents;
```

Comment:
Make the fields final, and instantiate inline
```suggestion
private final ImmutableList.Builder<QueryCreatedEvent> queryCreatedEvents = ImmutableList.builder();
        private final ImmutableList.Builder<QueryCompletedEvent> queryCompletedEvents = ImmutableList.builder();
        private final ImmutableList.Builder<SplitCompletedEvent> splitCompletedEvents = ImmutableList.builder();

        public EventsCapture()
        {
        }
```

Reply:
Done